tuxemon/states/combat/combat.py

```diff
@@ -473,6 +473,9 @@ def transition_phase(self, phase: CombatPhase) -> None:
             self.players[0].set_party_status()
             self.end_combat()
 
+            for monster, player in zip(self.active_monsters, self.active_players):
```

Comment:
Actually, I think we should just do this for the player, because modders might not want this for NPCs, and we don't save npc changes, so the evolution will disappear when you reload the game.

Reply:
If I remove monster (after for, leaving only player), it starts asking me:
```
    self.evolve(player, monster)
UnboundLocalError: local variable 'monster' referenced before assignment
```

Reply:
I'm going to try a thing, let's see if works

Reply:
Ideally you would track each monster in the players party who gained a level in the battle and evolve them.

If this is too difficult you could just try to evolve every monster in the party.

Reply:
It's the zip